### DISCUSSION TITLE
Server not reached on docker

### DISCUSSION THREAD

**Discussion by go-aegian** [connectrpc/connect-go#550] [Q&A]
When running the server on local the client connect-go works but as soon as I start the server in docker, I'm getting unavailable: EOF error.

The server url is the same in both scenarios, and other docker containers are working fine.

The code snippet that fails is located in protocol_connect.go file

```
Call chain 

func (m *connectUnaryMarshaler) Marshal(message any) *Error {
        ...       
	if len(data) < m.compressMinBytes || m.compressionPool == nil {
               	 ....
		**return m.write(data)**
	}
}

func (m *connectUnaryMarshaler) write(data []byte) *Error {
	if _, err := m.writer.Write(data); err != nil {
		...
	}
	return nil
}

This is where it fails as ClosedPipe error

func (d *duplexHTTPCall) Write(data []byte) (int, error) {
	...
	if err != nil && errors.Is(err, io.ErrClosedPipe) {
   	        ...
		**return bytesWritten, io.EOF**
	}
	return bytesWritten, err
}
```

Any suggestion of setting the client with some parameters for docker specifically?

**Comment by akshayjshah** [comment]
It's hard to debug this without a reproducible example. We use Connect extensively in Docker - especially in the [conformance tests](https://github.com/connectrpc/conformance) - and haven't run into problems.

Can you share more details?
